<launch>
    
    <arg name="camera_name" default="cam1" />
    <arg name="tracking_target" default="person" />

    <!--  
        Object pose estimation
    -->
    <node name="pengo_object_pose_estimation_node" pkg="hupster_detection" 
          type="hupster_object_pose_estimation_node" respawn="true" >
        <remap from="/camera/aligned_depth_to_color/image_raw" to="/$(arg camera_name)/aligned_depth_to_color/image_raw" />
        <remap from="/camera/color/camera_info" to="/$(arg camera_name)/color/camera_info" />
    </node>

    <node name="pengo_navigation_node" pkg="hupster_detection"
          type="person_follower_node">

        <param name="target" value="$(arg tracking_target)"/>
        <param name="min_distance" value="1.5" />
        <param name="max_distance" value="3.0" />

        <!-- Seconds -->
        <param name="tracking_timeout" value="2.0" />
        <param name="max_speed" value="0.8" />
        <param name="min_speed" value="0.2" />
        <param name="steering_factor" value="0.04" />
        <param name="enable" value="true" />
        <param name="base_frame" value="base_link" />
        <remap from="mobile_base/commands/velocity" to="navigation_velocity_smoother/raw_cmd_vel" />
        
    </node>

</launch>
